### ISSUE TITLE
java.lang.StackOverflowError when running update sql without WITH

### ISSUE THREAD

**Issue by grinapo** [dbeaver/dbeaver#12641]
#### System information: 
- Operating system (distribution) and version: Debian sid
- DBeaver version:  dbeaver-ce     21.0.5       amd64 
- Additional extensions

> Start DBeaver Standalone [org.jkiss.dbeaver.ui.app.standalone 21.0.5.202105160657]
> Start Eclipse e4 Workbench [org.eclipse.e4.ui.workbench 1.12.100.v20210122-1731]
> Start Eclipse IDE UI Application [org.eclipse.ui.ide.application 1.4.0.v20210122-1423]
> Start Eclipse Workbench Model [org.eclipse.e4.ui.model.workbench 2.1.1000.v20210111-0958]
> Start Eclipse Jobs Mechanism [org.eclipse.core.jobs 3.10.1100.v20210111-0815]
> Start Eclipse IDE UI [org.eclipse.ui.ide 3.18.100.v20210122-1536]
> Start DBeaver Core Plug-in [org.jkiss.dbeaver.core 21.0.5.202105160657]

> 2021-05-28 09:50:59.155 - OS: Linux 5.8.6 (amd64)
> 2021-05-28 09:50:59.155 - Java version: 11.0.9.1 by AdoptOpenJDK (64bit)

#### Connection specification:
- Database name and version: psql Debian 11.10-0+deb10u1
- Driver name: PostgreSQL JDBC Driver
- Do you use tunnels or proxies (SSH, SOCKS, etc)? no

#### Describe the problem you're observing:

opened sql notebook, entered
```SQL
UPDATE foo SET bar='2000-01-01';
```
and pressed `CTRL-ENTER`. Got (handled) crash:

```
2021-05-28 09:43:38.346 - Event loop exception
java.lang.StackOverflowError
	at java.base/java.util.ResourceBundle.getObject(Unknown Source)
	at java.base/java.util.ResourceBundle.getString(Unknown Source)
	at org.eclipse.jface.util.Util.translateString(Util.java:436)
	at org.eclipse.jface.bindings.keys.formatting.AbstractKeyFormatter.format(AbstractKeyFormatter.java:87)
	at org.eclipse.jface.bindings.keys.formatting.NativeKeyFormatter.format(NativeKeyFormatter.java:110)
	at org.eclipse.jface.bindings.keys.formatting.AbstractKeyFormatter.format(AbstractKeyFormatter.java:121)
	at org.eclipse.jface.bindings.keys.formatting.AbstractKeyFormatter.format(AbstractKeyFormatter.java:100)
	at org.eclipse.jface.bindings.keys.KeySequence.format(KeySequence.java:221)
	at org.eclipse.e4.ui.bindings.internal.BindingTable$BindingComparator.compare(BindingTable.java:92)
	at org.eclipse.e4.ui.bindings.internal.BindingTable$BindingComparator.compare(BindingTable.java:1)
	at java.base/java.util.TimSort.countRunAndMakeAscending(Unknown Source)
	at java.base/java.util.TimSort.sort(Unknown Source)
	at java.base/java.util.Arrays.sort(Unknown Source)
	at java.base/java.util.ArrayList.sort(Unknown Source)
	at org.eclipse.e4.ui.bindings.internal.BindingTableManager.getSequencesFor(BindingTableManager.java:189)
	at org.eclipse.e4.ui.bindings.internal.BindingTableManager.getBestSequenceFor(BindingTableManager.java:166)
	at org.eclipse.e4.ui.bindings.internal.BindingServiceImpl.getBestSequenceFor(BindingServiceImpl.java:136)
	at org.eclipse.ui.internal.keys.BindingService.getBestActiveBindingFor(BindingService.java:166)
	at org.eclipse.ui.menus.CommandContributionItem.updateMenuItem(CommandContributionItem.java:511)
	at org.eclipse.ui.menus.CommandContributionItem.update(CommandContributionItem.java:484)
	at org.eclipse.ui.menus.CommandContributionItem.lambda$2(CommandContributionItem.java:323)
	at org.eclipse.ui.menus.CommandContributionItem.updateCommandProperties(CommandContributionItem.java:327)
	at org.eclipse.ui.menus.CommandContributionItem.lambda$1(CommandContributionItem.java:294)
	at org.eclipse.core.commands.Command$1.run(Command.java:529)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.commands.Command.fireCommandChanged(Command.java:522)
	at org.eclipse.core.commands.Command.lambda$0(Command.java:1002)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.fireHandlerChanged(HandlerServiceHandler.java:180)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.handlerChanged(E4HandlerProxy.java:115)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.ui.internal.handlers.HandlerProxy.lambda$1(HandlerProxy.java:372)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.core.commands.AbstractHandler.setBaseEnabled(AbstractHandler.java:111)
	at org.eclipse.ui.texteditor.AbstractTextZoomHandler.setEnabled(AbstractTextZoomHandler.java:131)
	at org.eclipse.ui.texteditor.TextZoomInHandler.setEnabled(TextZoomInHandler.java:1)
	at org.eclipse.ui.internal.handlers.HandlerProxy.setEnabled(HandlerProxy.java:229)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.setEnabled(E4HandlerProxy.java:133)
	at jdk.internal.reflect.GeneratedMethodAccessor26.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:319)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:253)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.setEnabled(HandlerServiceHandler.java:84)
	at org.eclipse.core.commands.Command.setEnabled(Command.java:856)
	at org.eclipse.ui.menus.CommandContributionItem.isEnabled(CommandContributionItem.java:916)
	at org.eclipse.ui.menus.CommandContributionItem.updateMenuItem(CommandContributionItem.java:531)
	at org.eclipse.ui.menus.CommandContributionItem.update(CommandContributionItem.java:484)
	at org.eclipse.ui.menus.CommandContributionItem.lambda$2(CommandContributionItem.java:323)
	at org.eclipse.ui.menus.CommandContributionItem.updateCommandProperties(CommandContributionItem.java:327)
	at org.eclipse.ui.menus.CommandContributionItem.lambda$1(CommandContributionItem.java:294)
	at org.eclipse.core.commands.Command$1.run(Command.java:529)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.commands.Command.fireCommandChanged(Command.java:522)
	at org.eclipse.core.commands.Command.lambda$0(Command.java:1002)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.fireHandlerChanged(HandlerServiceHandler.java:180)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.handlerChanged(E4HandlerProxy.java:115)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.ui.internal.handlers.HandlerProxy.lambda$1(HandlerProxy.java:372)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.core.commands.AbstractHandler.setBaseEnabled(AbstractHandler.java:111)
	at org.eclipse.ui.texteditor.AbstractTextZoomHandler.setEnabled(AbstractTextZoomHandler.java:131)
	at org.eclipse.ui.texteditor.TextZoomInHandler.setEnabled(TextZoomInHandler.java:1)
	at org.eclipse.ui.internal.handlers.HandlerProxy.setEnabled(HandlerProxy.java:229)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.canExecute(E4HandlerProxy.java:76)
	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[...repeat many times...]
```
and the end was like
```
	at jdk.internal.reflect.GeneratedMethodAccessor26.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:319)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:253)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.setEnabled(HandlerServiceHandler.java:84)
	at org.eclipse.core.commands.Command.setEnabled(Command.java:856)
	at org.eclipse.ui.menus.CommandContributionItem.isEnabled(CommandContributionItem.java:916)
	at org.eclipse.ui.menus.CommandContributionItem.updateMenuItem(CommandContributionItem.java:531)
	at org.eclipse.ui.menus.CommandContributionItem.update(CommandContributionItem.java:484)
	at org.eclipse.ui.menus.CommandContributionItem.lambda$2(CommandContributionItem.java:323)
	at org.eclipse.ui.menus.CommandContributionItem.updateCommandProperties(CommandContributionItem.java:327)
	at org.eclipse.ui.menus.CommandContributionItem.lambda$1(CommandContributionItem.java:294)
	at org.eclipse.core.commands.Command$1.run(Command.java:529)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.commands.Command.fireCommandChanged(Command.java:522)
	at org.eclipse.core.commands.Command.lambda$0(Command.java:1002)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.fireHandlerChanged(HandlerServiceHandler.java:180)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.handlerChanged(E4HandlerProxy.java:115)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.ui.internal.handlers.HandlerProxy.lambda$1(HandlerProxy.java:372)
	at org.eclipse.core.commands.AbstractHandler.fireHandlerChanged(AbstractHandler.java:77)
	at org.eclipse.core.commands.AbstractHandler.setBaseEnabled(AbstractHandler.java:111)
```

**Comment by serge-rider** [comment]
Probably this was fixed in 21.1 version (in d587cfd1d1c1b5c94b50a1a04cc17ac7a7602f01).

Could you test it in the Early Access version (https://dbeaver.io/files/ea)?

**Comment by grinapo** [comment]
> Probably this was fixed in 21.1 version (in [d587cfd](https://github.com/dbeaver/dbeaver/commit/d587cfd1d1c1b5c94b50a1a04cc17ac7a7602f01)).
> 
> Could you test it in the Early Access version (https://dbeaver.io/files/ea)?

It does not enter an infinite loop in 21.1.0.202105250725, I'd say it can be safely closed. Thank you!
